mreg/mixins.py

```diff
@@ -18,4 +44,28 @@ def get_object(self):
         # May raise a permission denied
         self.check_object_permissions(self.request, obj)
 
-        return obj
+        return obj        
+
+    def get_object_from_request(self: DetailViewProtocol, request: Request) -> Union[Any, None]:
+        """Return the object defined by the key lookup_field in request.data, if any.
+
+        Note: This is part of the LowerCaseLookupMixin, so the value of the lookup_field
+        in request.data will be lowercased.
+
+        :param request: The request object.
+
+        :returns: The object from the queryset or None.
+        """
+        if not self.lookup_field:
+            raise AttributeError("lookup_field must be defined.")
+        
+        if not request.data or not isinstance(request.data, QueryDict):
```

Comment:
```suggestion
        if not request.data or not isinstance(request.data, dict):
```

Otherwise we lose JSON support here.